webdriver-spec.html

```diff
@@ -5148,35 +5154,64 @@
    <li><p>Remove all <var>undo actions</var> from the <a>current
     session</a>'s <a>input cancel list</a>.
 
-  <li><p>Let <var>i</var> be 0.
-
-  <li><p>Let <var>length</var> be the length of <var>character array</var>.
-
-  <li><p>While <var>i</var> &lt; <var>length</var>:
-
-    <ol>
-      <li><p>Let <var>char</var> be the entry in <var>character array</var> at
-        index <var>i</var>.
-
-      <li><p>Let <var>key to type</var> be the result of <a>processing keys</a>
-        with argument <var>key</var>.
+   <li><p>Remove all <a>key input source</a>s from the <a>current
+    session</a>'s <a>input state</a>.
 
-      <li><p>If the <var>key to type</var> is a <a>modifier key</a> then set the relevant
-        attribute on <a>Keyboard Event</a> to true for the rest of this loop.
+   <li><p><a>Create a new key input state
+    object</a>. Let <var>keyboard</var> equal this newly
+    created <a>key input state</a>.
```

Comment:
We don’t have to be explicit about the creating a new key input state object since it’s implicitly created when creating a _key input source_: https://w3c.github.io/webdriver/webdriver-spec.html#dfn-key-input-source